simtools/applications/plot_array_layout.py

```diff
@@ -122,106 +182,267 @@ def _get_site_from_telescope_list_name(telescope_list_file):
+    """
+    Get list of output file names for plotting.
 
-    if args_dict["rotate_angle"] is None:
-        rotate_angles = [0 * u.deg]
-    else:
-        rotate_angles = []
-        for one_angle in args_dict["rotate_angle"]:
-            rotate_angles.append(float(one_angle) * u.deg)
+    Parameters
+    ----------
+    plot_file_name : str
+        Name of the plot file.
+    output_dir : str
+        Output directory.
 
-    logger = logging.getLogger()
-    logger.setLevel(gen.get_log_level_from_user(args_dict["log_level"]))
-    mpl.use("Agg")
-    telescope_file = None
-    if args_dict["telescope_list"] is not None:
-        logger.info("Plotting array from telescope list file(s).")
-        telescope_file = args_dict["telescope_list"]
-
-    elif args_dict["array_layout_name"] is not None:
-        logger.info("Plotting array from layout array name(s).")
-        telescope_file = [
-            io_handler_instance.get_input_data_file(
-                "layout", f"telescope_positions-{one_array}.ecsv"
-            )
-            for one_array in args_dict["array_layout_name"]
-        ]
-
-    for one_file in telescope_file:
-        logger.debug(f"Processing: {one_file}.")
+    Returns
+    -------
+    list
+        List of output file names.
+
+    Raises
+    ------
+    NameError
+        If the file extension is not valid.
+    """
+    plot_file = output_dir.joinpath(plot_file_name)
+
+    if len(plot_file.suffix) == 0:
+        return [plot_file.with_suffix(f".{ext}") for ext in ["pdf", "png"]]
+
+    allowed_extensions = [".jpeg", ".jpg", ".png", ".tiff", ".ps", ".pdf", ".bmp"]
+    if plot_file.suffix in allowed_extensions:
+        return [plot_file]
+    msg = f"Extension in {plot_file} is not valid. Valid extensions are:" f" {allowed_extensions}."
+    raise NameError(msg)
+
+
+def _get_plot_file_name(figure_name, layout_name, site, coordinate_system, rotation_angle):
+    """
+    Generate and return the file name for plots.
+
+    Parameters
+    ----------
+    figure_name : str
+        Figure name given through command line.
+    layout_name : str
+        Name of the layout.
+    site : str
+        Site name.
+    coordinate_system : str
+        Coordinate system for the array layout.
+    rotation_angle : float
+        Angle to rotate the array before plotting.
+
+    Returns
+    -------
+    str
+        Plot file name.
+    """
+    if figure_name is not None:
+        return figure_name
+
+    return (
+        f"array_layout_{layout_name}_{site}_{coordinate_system}_"
+        f"{str(round(rotation_angle.to(u.deg).value))}deg"
+    )
+
+
+def _layouts_from_array_layout_file(args_dict, db_config, rotate_angle):
+    """
+    Read array layout positions from file(s) and return a list of layouts.
+
+    Parameters
+    ----------
+    args_dict : dict
+        Dictionary with the command line arguments.
+    db_config : dict
+        Database configuration.
+    rotate_angle : float
```

Comment:
`rotation_angle` to be consistent with previous naming

Reply:
I've looked now through simtools and we are using almost everywhere `rotate_angle`. I therefore changed all `rotation_angle` to `rotate_angle`.